data_collection/gazette/utils/text_extraction.py

```diff
@@ -0,0 +1,138 @@
+    string, month in full writing in pt-br, commonly used separators as '-', '/',
+    'de', ' ' or without it). Adds ways to deal with inconsistent situations,
+    such as when there's only one separator in text, ordenation symbol (ªº) present
+    or mistyped month. Check test_dates.py in tests/ directory to see examples.
+
+    By default, it's set to do stric_parsing, meaning it doesn't try to parse
+    incomplete data, and only handles date in day month year order (DMY) for now,
+    because its enough for all using cases (but an improvement can be implemented
+    in the future, on demand)
+
+    Args:
+        text: string with date text in it
+    Returns:
+        Date object, if able to parse. Otherwise, returns None.
+    """
+    if not isinstance(text, str):
+        return None
+
+    order = "DMY"
+    settings = {"DATE_ORDER": order, "STRICT_PARSING": True}
+
+    # normalize input
+    text = re.sub("[°ºª]", "", text)
+    normalized_text = unidecode(re.sub(" +", " ", text).strip().lower())
+
+    # To avoid issues with dirty strings, this function uses full concatenated text preferably
+    # 1nov2025 > month behaves as separator, allowing numbers identification
+    # 1112025 > numbered str might has ambiguity (it is 1/11 or 11/1 or 11/12/025?)
+    concatenated_text = re.sub(" ", "", normalized_text)
+
+    # HANDLES FULL NUMBERED LIMITATIONS...
+    is_full_numbered = concatenated_text.isdigit()
+    has_space_separators = sum(char == " " for char in normalized_text) == 2
+    if is_full_numbered:
+        # 1- space is a separator. Concatenation will ruin its parse
+        if has_space_separators:  # D M YYYY | D MM YYYY | DD M YYYY | DD MM YYYY
+            parsed_date = parse(normalized_text, languages=["pt"], settings=settings)
+            if parsed_date:
+                return parsed_date.date()
+        # 2- there is no separator. Only possible to parse when it has 8 digits
+        if len(concatenated_text) != 8:  # exactly DDMMYYYY
+            return None
+
+    # ... FOLLOWS DEFAULT FLOW (with full concatenated string)
+
+    # replace 'de' separator, without conflict with 'de' of 'dezembro'
+    # note: makes 'de' cases behave as if DD - MM - AAAA
+    text = concatenated_text.replace("de", "-").replace("-z", "dez")
+
+    # TRY TO APPLY DATEPARSER PARSING
+    parsed_date = parse(text, languages=["pt"], settings=settings)
+    if parsed_date:
+        return parsed_date.date()
+
+    # TRY TO APPLY REGEX PARSING
+
+    # create order sensibility.
+    regex_date_order = ["", "", ""]
+
+    day_index = order.index("D")
+    month_index = order.index("M")
+    year_index = order.index("Y")
+
+    regex_date_order[day_index] = r"(\d{1,2})"
+    regex_date_order[month_index] = r"(\d{1,2}|[a-z]+)"
+    regex_date_order[year_index] = r"(\d{4})"
+
+    # builds a complete search regex, like
+    # (D or M or Y) separators (D or M or Y) separators (D or M or Y)
```

Comment:
Esse comentário indica que o regex final encontra datas com partes (D or M or Y) em qualquer ordem, mas não é o caso, certo?
É esperado uma ordem fixa definida em `order`

Talvez seja melhor se ele dissesse algo como `# builds a complete search regex with different separators and ways of writing the date parts, in the order defined by 'order'`

Reply:
Isso. Melhorei isso abstraindo para _apply_regex() essa questão de flexibilidade. Por enquanto, só DMY foi necessário, mas ja vi casos que a ordem é outra. O dateparser tem um recurso nativo de indicar a ordem, mas o regex é muito "rigido", vai ser a ordem da string do padrão. Queria deixar o recurso (de poder indicar qual a ordem) meio pré-pensado para ir incrementando.